endpoints/openrtb2/auction.go

```diff
@@ -1391,22 +1394,47 @@ func (deps *endpointDeps) processStoredRequests(ctx context.Context, requestJson
 					err = fmt.Errorf("imp.ext.prebid.storedrequest.id %s: Stored Imp has Invalid JSON: %s", impIds[i], Err)
 				}
 			}
-			return nil, []error{err}
+			return nil, impToStoredReq, []error{err}
 		}
 		imps[idIndices[i]] = resolvedImp
+
+		includeStoredImps, err := jsonparser.GetBoolean(resolvedImp, "ext", "prebid", "options", "echovideoattrs")
+
+		if err != nil && err != jsonparser.KeyPathNotFoundError {
+			return nil, nil, []error{err}
+		}
+
+		//determine if this is video impression
+		_, _, offset, err := jsonparser.Get(resolvedImp, "video")
+		if err != nil && err != jsonparser.KeyPathNotFoundError {
+			return nil, nil, []error{err}
+		}
+
+		if includeStoredImps && offset != -1 {
+			impId, err := jsonparser.GetString(resolvedImp, "id")
+			if err != nil {
+				return nil, nil, []error{err}
+			}
+			//extract video attributes only
+			videoData, _, _, err := jsonparser.Get(storedImps[impIds[i]], "video")
+			if err == nil {
+				impToStoredReq[impId] = videoData
```

Comment:
The name `impToStoredReq` is now a bit misleading since it's a subset for video auctions. Perhaps it would be best to include the full stored request here and "cut out" the video section to bid response handler?

Reply:
I went through this again. The reason I put this here is to optimize amount of data we have in memory (have only video impressions we will use later) and to reduce complexity of code execution in `insertStoredImpData` function: if no stored video data found - do not execute any logic. Also if we leave this code here - it will fit the logic around it: prepare stored request/video data. 
I'm open to modify this and move this piece of code to bid response handler (insertStoredImpData). Please advice.

Reply:
Renamed `impToStoredReq` to `impToStoredVideoReq`

Reply:
Modified according to discussion